### PR TITLE
docs: Get rid of badly named readme

### PR DESCRIPTION
**Pull request by merland**
With its current name, the file `doc/README_osx.md` looks like an entry point README for OSX users, but it only contains specific instructions on how to build a DMG. 

This PR deletes the file and moves the contents of the file into `doc/build-osx.md`.

### CODE REVIEW

**Comment by laanwj** [comment]
As these are build instructions, maybe it would make sense to integrate it into the existing `build_osx.md`?
I don't think the `README_` prefix makes much sense in the documentation directory.

**Comment by merland** [comment]
@laanwj I was thinking the same, but since `doc/release-process.md` links to the DMG instructions, it could make sense to keep them in a separate file. Alternatively link to an anchor inside `build_osx.md`.

**Comment by promag** [comment]
NACK renaming to _dmg.

Even README_windows.txt` could be removed and integrated into doc/README.md#Windows

**Comment by laanwj** [comment]
Well the point of README_windows.txt is as an introductionary text for windows users. That one makes sense. It's also copied into the windows archives.

@merland's point is that this text is not an introductionary text for OSX users so is named wrongly.

**Comment by merland** [comment]
Regarding `README_windows.txt` I also argue for its removal due to out-of-context info and duplication. I actually tried to remove it before (#14755) but that failed the Gitian build. Anyway, I think it would be better to handle that in a separate PR.

**Comment by laanwj** [comment]
> Regarding README_windows.txt I also argue for its removal, due to out-of-context info and duplication

It's actually used! so no, I'd argue not to remove it! Yes, there is some duplication, but also some things that are different for windows. If you'd remove it you'd have to replace it with something else similar anyhow (which is used as an introductionary text for windows users). Feel free to rewrite it though.

**Comment by merland** [comment]
@laanwj Yes, I know the file is used but I have a feeling we can find a better strategy. One that satisfies both those who read the archive and those who read from GitHub, with no build script repackaging.

My best idea for a compromise would be to keep the file but have it contain only a reference to the windows anchor in the main `doc/README.md`. Thoughts on that?

Related: #15161

**Comment by laanwj** [comment]
No, that won't work because the `doc/README.md` is not copied in the same way!
`doc/README.md` is the Table of Contents of the `doc` file. I guess that would work *if* the entire folder is copied, but it'd also be confusing to new users because that folder contains user documentation.

So the file should stand on its own. I think it's fine, let's just not change this okay.

**Comment by merland** [comment]
Ok, there is obviously something I'm missing here. Where can I learn more about how the docs are copied/packaged?

**Comment by promag** [comment]
See

https://github.com/bitcoin/bitcoin/blob/d44b01f028e27c02a026ba1470d631ded51e07c1/Makefile.am#L54-L57

https://github.com/bitcoin/bitcoin/blob/d44b01f028e27c02a026ba1470d631ded51e07c1/share/setup.nsi.in#L73-L79

**Comment by merland** [comment]
Updated commit, title and description. Please re-review!  

(Note that the lengthy discussion about `README_windows.txt` above can be ignored completely in the context of the proposed changes.)

File: doc/release-process.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ Ensure gitian-builder is up-to-date:
     echo 'f9a8cdb38b9c309326764ebc937cba1523a3a751a7ab05df3ecc99d18ae466c9 inputs/osslsigncode-1.7.1.tar.gz' | sha256sum -c
     popd
 
-Create the macOS SDK tarball, see the [macOS readme](README_osx.md) for details, and copy it into the inputs directory.
+Create the macOS SDK tarball, see the [OSX build instructions](build-osx.md#deterministic-macos-dmg-notes) for details, and copy it into the inputs directory.
```
[the remark this answers is not in the document]
Comment (promag) [reply doc/release-process.md]: I mean you should keep macOS.

[the remark this answers is not in the document]
Comment (merland) [reply doc/release-process.md]: Ah, I see it now. Will update.

[the remark this answers is not in the document]
Comment (merland) [reply doc/release-process.md]: updated

**Comment by laanwj** [comment]
utACK f24ed6d39f963e7f1c0b4dbd9b784132ef975f2a

**Pull request merged**